Add connector and forwarder examples (#3202)
EnMasseProject/enmasse: documentation/modules/ref-address-example-forwarders.adoc

// Module included in the following assemblies:
//
// assembly-managing-addresses.adoc

[id='ref-address-example-forwarders-{context}']
== Address forwarding examples

Forwarders are a way to automatically forward messages from an 'local' address to a remote AMQP address outside of {ProductName}, or forward messages from a remote AMQP address to a 'local' address. Creating a forwarder requires that a connector to the remote AMQP endpoint has been configured for the address space. For more information on connectors, see link:{BookUrlBase}{BaseProductVersion}{BookNameUrl}#ref-address-space-example-connectors-messaging[Address space connector examples].

Forwarding is only supported in the `standard` address space type, and only for the `queue` and `subscription` address type. The `subscription` address type only supports forwarding in the `out` direction.

> These examples expect that a connector `remote` has been configured for the address space.

Author: If the procedure links to the connector ref, I think that would be good. The examples are made to 'work together'.

----
apiVersion: enmasse.io/v1beta1
kind: Address
metadata:
  name: myspace.myqueue
spec:
  address: myqueue
  type: queue
  plan: standard-small-queue
  forwarders:
  - name: f1 <1>
    remoteAddress: remote1/clients/me/1 <2>
    direction: out <3>
----
<1> (Required) The name of the forwarders. This is used to ensure a unique identity of the forwarder.
<2> (Required) The remote address to forward messages to. The address must be prefixed with the connector name, and match the address matching patterns defined on the connector.
<3> (Required) The direction of message flow, either `out` or `in`. `out` will forward messages to the remote endpoint. `in` will forward from the remote endpoint.

== Forwarding messages from a remote endpoint to a local queue

In this example, the remote endpoint has an address `prices/milk` that we can receive messages from and put in `myqueue`.

[source,yaml,options="nowrap"]
----
apiVersion: enmasse.io/v1beta1
kind: Address
metadata:
  name: myspace.myqueue
spec:
  address: myqueue
  type: queue
  plan: standard-small-queue
  forwarders:
  - name: f1 <1>
    remoteAddress: remote1/prices/milk <2>
    direction: in <3>
----

> <1> (Required) The name of the forwarders. This is used to ensure a unique identity of the forwarder.

Reviewer: ```suggestion
<1> (Required) Specifies the name of the forwarder, which is used to ensure a unique identity.
```

> <2> (Required) The remote address to forward messages to. The address must be prefixed with the connector name, and match the address matching patterns defined on the connector.

Reviewer: ```suggestion
<2> (Required) Specifies the remote address to forward messages to. The address must be prefixed with the connector name and must be identical to the address matching patterns defined on the connector.
```

> <3> (Required) The direction of message flow, either `out` or `in`. `out` will forward messages to the remote endpoint. `in` will forward from the remote endpoint.

Reviewer: ```suggestion
<3> (Required) Specifies the direction of message flow, which is either `out` or `in`. A value of `out` forwards messages to the remote endpoint. A value of `in` forwards messages from the remote endpoint.
```
